src/main/scala/workflow/graph/ExtractSaveablePrefixes.scala

```diff
@@ -0,0 +1,19 @@
+package workflow.graph
+
+/**
+ * Extract the prefixes whose state we want to save (aka for any cacher or estimator node)
```

Comment:
This basically looks ok. It kind of seems like this set of rules is somehow special?